File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const new_lost = lost.filter(v => !reserve.includes(v)).sort((a, b) => a - b);
    const new_reserve = reserve.filter(v => !lost.includes(v)).sort((a, b) => a - b);
    let answer = n - new_lost.length;
    
    new_lost.forEach(target => {
        const a = new_reserve.indexOf(target - 1);
        const b = new_reserve.indexOf(target + 1);

        if (a >= 0) {
            new_reserve.splice(a, 1);
            answer++;
        }
        else if (b >= 0) {
            new_reserve.splice(b, 1);
            answer++;
        }
    });
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    new_lost = sorted([v for v in lost if v not in reserve])
    new_reserve = sorted([v for v in reserve if v not in lost])
    answer = n - len(new_lost)

    for target in new_lost:
        a = target - 1
        b = target + 1

        if a in new_reserve:
            new_reserve.remove(a)
            answer += 1
        elif b in new_reserve:
            new_reserve.remove(b)
            answer += 1

    return answer
